pkg/cloudprovider/providers/gce/gce_networkendpointgroup.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"strings"
+	"time"
+
+	computealpha "google.golang.org/api/compute/v0.alpha"
+)
+
+const (
+	ExperimentalKey      = "X-Goog-Experiments"
+	NEGExperimentalValue = "EnableNetworkEndpointGroup"
+)
+
+func newNetworkEndpointGroupMetricContext(request string, zone string) *metricContext {
+	zoneLabel := zone
+	if len(strings.TrimSpace(zone)) == 0 {
+		zoneLabel = unusedMetricLabel
+	}
+	return &metricContext{
+		start:      time.Now(),
+		attributes: []string{"networkendpointgroup_" + request, unusedMetricLabel, zoneLabel},
+	}
+}
+
+func (gce *GCECloud) GetNetworkEndpointGroup(name string, zone string) (*computealpha.NetworkEndpointGroup, error) {
+	if err := gce.alphaFeatureEnabled(AlphaFeatureNetworkEndpointGroup); err != nil {
+		return nil, err
+	}
+	mc := newNetworkEndpointGroupMetricContext("get", zone)
+	v, err := gce.serviceAlpha.NetworkEndpointGroups.Get(gce.GetProjectID(), zone, name).Do()
+	return v, mc.Observe(err)
+}
+
+func (gce *GCECloud) ListNetworkEndpointGroup(zone string) (*computealpha.NetworkEndpointGroupList, error) {
+	if err := gce.alphaFeatureEnabled(AlphaFeatureNetworkEndpointGroup); err != nil {
+		return nil, err
+	}
+	mc := newNetworkEndpointGroupMetricContext("list", zone)
+	v, err := gce.serviceAlpha.NetworkEndpointGroups.List(gce.GetProjectID(), zone).Do()
+	return v, mc.Observe(err)
+}
+
+func (gce *GCECloud) AggregatedListNetworkEndpointGroup() (*computealpha.NetworkEndpointGroupAggregatedList, error) {
+	if err := gce.alphaFeatureEnabled(AlphaFeatureNetworkEndpointGroup); err != nil {
+		return nil, err
+	}
+	mc := newNetworkEndpointGroupMetricContext("aggregated", "")
+	v, err := gce.serviceAlpha.NetworkEndpointGroups.AggregatedList(gce.GetProjectID()).Do()
+	return v, mc.Observe(err)
+}
+
+func (gce *GCECloud) CreateNetworkEndpointGroup(name string, zone string) error {
+	if err := gce.alphaFeatureEnabled(AlphaFeatureNetworkEndpointGroup); err != nil {
+		return err
+	}
+	mc := newNetworkEndpointGroupMetricContext("create", zone)
+	networkEndpointGroup := &computealpha.NetworkEndpointGroup{
+		Name:                name,
+		Type:                "LOAD_BALANCING",
+		NetworkEndpointType: "GCE_VM_IP_PORT",
+		LoadBalancer: &computealpha.NetworkEndpointGroupLbNetworkEndpointGroup{
+			Network:    gce.networkURL,
+			Subnetwork: gce.subnetworkURL,
+		},
+	}
```

Comment:
It is just simpler. As the only variances are `name` and `zone`. I can switch to the other way.